event/event.go

```diff
@@ -28,16 +28,18 @@ type (
 		Event   T      `json:"event"`
 	}
 
+	// Subscription is a subscription that received only specific types of events
+	// defined by T.
 	Subscription[T any] struct {
 		name   string
 		rawsub *MsgSubscription
 	}
 
-	// EventHandler is responsible for handling events from a subscription.
+	// SubscriptionHandler is responsible for handling events from a [Subscription].
```

Comment:
You didn't want to just call it `Handler` (making it `event.Handler`)?

Reply:
I thought about it, it was my first option, but we also have the MsgSubscription and its Handler...so I was on the fence if it was not going to be too vague.

Reply:
Like on event we have Publisher/Subscription/MsgSubscription...it felt like Handler would not be specific enough...but it does read very well =P

Reply:
WDYT @guilherme-santos ?

Reply:
Yeah, I like how it reads as well. Maybe MsgSubscription is not great.

As the struct is called `Message`, `Handler` and `MessageHandler` wouldn't be good?

Reply:
hmm yeah, so it would be MessageHandler => Message and Handler -> event.Envelope[T] ?

Reply:
hmmm I really like just event.Handler and it would work perfectly without the Message stuff.... will go with this name and hope we can remove the old stuff someday :laughing: